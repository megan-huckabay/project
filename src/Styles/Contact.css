.contact-container {
  border: 5px solid black;
  margin: 5px;
  padding: 5px;
  background-color: #b894f1; /* Adjusted to a shade of purple (indigo) */
  text-align: center;
  font-weight: bold;
  font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
}

.contact-page p {
  color: #000000;
}

.contact-page a {
  color: #f1f1f1; /* Adjusted to a shade of green */
}

.contact-page h3 {
  color: #0cc9d6;
  text-shadow: 8px 4px 15px #bdcaca;
}
