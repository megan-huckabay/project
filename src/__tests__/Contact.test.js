import 'jest-styled-components';
import React from 'react';
import { render, screen } from '@testing-library/react';
import Contact from '../Components/Contact';

test('renders the Contact component and checks links', () => {
  render(<Contact />);
  
  // Test component rendering
  const contactTitle = screen.getByText(/Contact Information/i);
  expect(contactTitle).toBeInTheDocument();

  // Test email links
  const mainEmailLink = screen.getByRole('link', { name: /techchick2023@gmail.com/i });
  expect(mainEmailLink).toHaveAttribute('href', 'mailto:techchick2023@gmail.com');

  const alternativeEmailLink = screen.getByRole('link', { name: /huckabay90@gmail.com/i });
  expect(alternativeEmailLink).toHaveAttribute('href', 'mailto:huckabay90@gmail.com');

  // Test GitHub link
  const githubLink = screen.getByRole('link', { name: /GitHub/i });
  expect(githubLink).toHaveAttribute('href', 'https://github.com/meggers1990');

  // Test LinkedIn link
  const linkedinLink = screen.getByRole('link', { name: /LinkedIn/i });
  expect(linkedinLink).toHaveAttribute('href', 'https://linkedin.com/in/megan-huckabay');

  // Test Resume link
  const resumeLink = screen.getByRole('link', { name: /Click for Resume/i });
  expect(resumeLink).toHaveAttribute('href', 'https://docs.google.com/document/d/1d84juiC488FZ55L0Wioiz5BCJ5KyngAjabOw17EoqIc/edit?usp=sharing');
});

test('matches snapshot', () => {
  const { container } = render(<Contact />);
  expect(container).toMatchSnapshot();
});
