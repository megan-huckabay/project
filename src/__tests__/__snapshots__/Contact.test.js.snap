// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="contact-container contact-page"
  >
    <h3>
      Contact Information
    </h3>
    <p>
      main-email: 
      <a
        href="mailto:techchick2023@gmail.com"
      >
        techchick2023@gmail.com
      </a>
      <br />
      alternative-email: 
      <a
        href="mailto:huckabay90@gmail.com"
      >
        huckabay90@gmail.com
      </a>
      <br />
      GitHub:  
      <a
        href="https://github.com/meggers1990"
      >
        GitHub
      </a>
      <br />
      LinkedIn: 
      <a
        href="https://linkedin.com/in/megan-huckabay"
      >
        LinkedIn
      </a>
      <br />
      Resume Link: 
      <a
        href="https://docs.google.com/document/d/1d84juiC488FZ55L0Wioiz5BCJ5KyngAjabOw17EoqIc/edit?usp=sharing"
      >
        Click for Resume
      </a>
    </p>
  </div>
</div>
`;
